<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>445</width>
    <height>577</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="title">
       <string>Connect Config</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>COM PORT:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="inputComPort">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
         <property name="text">
          <string>COM1</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Baudrate:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="inputBaudrate">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
         <property name="text">
          <string>38400</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnConnect">
         <property name="styleSheet">
          <string notr="true">selection-background-color: rgb(255, 255, 102);
background-color: rgb(235, 235, 235);</string>
         </property>
         <property name="text">
          <string>connect</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Read Data</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>function</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>address</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>number</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnReadInputRegs">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>Read Input Registers</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="spinReadInputAddr">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="spinReadInputNum">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="btnReadHoldingRegs">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>Read Holding Registers</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSpinBox" name="spinReadHoldingAddr">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="spinReadHoldingNum">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Write Data</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>function</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>address</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>value</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnWriteSingleReg">
         <property name="minimumSize">
          <size>
           <width>140</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>Write Single Register</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="spinWriteAddr">
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLineEdit" name="inputWriteValue">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>45</width>
           <height>24</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QPlainTextEdit" name="txtDisp"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>445</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
